### PR TITLE
etcdserver: add timeout param on getClusterFromRemotePeers

### LINKED ISSUE #3215: proxy fails to work when latency to active member is > 500ms
From Min Ren:
```
Hi,

We were trying to simulate the environment where our proxy mode node has large latency to connect with active nodes. When we added about 500ms delay, we are seeing this error. 

015/07/31 23:51:49 etcdserver: could not get cluster response from http://10.9.15.193:7001: Get http://10.9.15.193:7001/members: read tcp 10.9.15.193:7001: use of closed network connection
2015/07/31 23:51:49 etcdmain: proxy: could not retrieve cluster information from the given urls
2015/07/31 23:51:50 proxy: zero endpoints currently available


We noticed that there is "proxy-dial-timeout" configuration for proxy mode,  which has default 1 sec.  But changing that doesn't make any better. 

Then we look through the code,  and found etcd using hard coded timeout 1sec in this code, 

Could you confirm that this is the reason and why it is hard coded?


 func getClusterFromRemotePeers(urls []string, logerr bool, tr *http.Transport) (*cluster, error) {
	cc := &http.Client{
		Transport: tr,
		Timeout:   time.Second,
	}
	for _, u := range urls {
		resp, err := cc.Get(u + "/members")
		if err != nil {
			if logerr {
				plog.Warningf("could not get cluster response from %s: %v", u, err)
			}
			continue
		}
```

It is a problem in etcd proxy. We are improving the workflow for high-latency enviroment now, and this needs to be fixed too.

### PR DESCRIPTION
**Pull request by yichengq**
It sets no timeout for public GetClusterFromRemotePeers.

This helps the following cases to work well in high latency scenario:

1. proxy sync members from the cluster
2. newly-joined member sync members from the cluster

After removing request timeout, the request is still controlled by dial
timeout and read connection timeout, which is 5s. This avoids
unreasonably long wait.

fixes #3215

### CODE REVIEW

File: etcdserver/cluster_util.go [hunk 1]
```diff
@@ -51,14 +51,16 @@ func isMemberBootstrapped(cl *cluster, member string, tr *http.Transport) bool {
 // a response, or a Cluster cannot be successfully created from a received
 // response, an error is returned.
 func GetClusterFromRemotePeers(urls []string, tr *http.Transport) (*cluster, error) {
-	return getClusterFromRemotePeers(urls, true, tr)
+	return getClusterFromRemotePeers(urls, 0, true, tr)
```
Comment (xiang90) [inline etcdserver/cluster_util.go]: let set 5 seconds or higher here. it is much readable and understandable than 0.

**Comment by yichengq** [comment]
set a reasonable timeout instead of no timeout. PTAL

**Comment by xiang90** [comment]
LGTM

**Comment by minjs** [comment]
Is this in 2.2.0 release?

**Comment by xiang90** [comment]
Yes

**Pull request merged**